src/room.js

```diff
@@ -0,0 +1,36 @@
+class Room {
+
+  constructor (roomname) {
+    this.roomname = roomname;
+    this.endpoints = {};
+  }
+  addEndpoint (endpointId) {
+    this.endpoints[endpointId] = {};
+  }
+  updateUsername (endpointId, username) {
+    this.endpoints[endpointId]['username'] = username;
```

Comment:
@esraajb You still need to have `[username]`, putting it in a string like `['username']` won't work. Otherwise you're looking for a user whose name is "username".

same for the other `update` methods